logging/logger.go

```diff
@@ -0,0 +1,77 @@
+package logging
+
+import (
+	"context"
+)
+
+var (
+	// DefaultLogger is the default Logger for all SDK logging, if one hasn't been provided in the context.
+	DefaultLogger Logger = &NoOpLogger{}
+
+	contextKey = loggerContextKey{}
+)
+
+type loggerContextKey struct{}
+
+// FromContext returns the Logger set in the context with Context(), or the DefaultLogger if no Logger is set in the context.
+// If DefaultLogger is nil, it returns a *NoOpLogger so that the return is always valid to call methods on without nil-checking.
+// So long as the Logger to return is not the NoOpLogger, it will call WithContext(ctx) on the logger in the context and return
```

Comment:
Do you think this could be rephrased to something like "So long as the Logger to return is not the NoOpLogger, it will always carry down the input context" (probably there's a much better way to phrase it)?
I believe this is more direct and doesn't require people to scroll down to `Logger` interface to check what `WithContext` function does.